Added windows installer files and makefile target.

aaron-prindle opened the pull request:
#449

dlorenc commented on Makefile:
```diff
@@ -17,7 +17,12 @@ export GO15VENDOREXPERIMENT=1
 # Bump these on release
 VERSION ?= v0.9.0
+# There's definitely a better way to do this, should I use cut from the VERSION?
+# Or should we have major, minor, and build and then construct VERSION/DEB_VERSION from that?
```
Yeah I like breaking it into major/minor/build and then constructing these from that.

dlorenc commented:
cc @iantalarico

Ian, could you take a loom here and let us know if we're on the right track? :)

iantalarico commented:
@dlorenc Of course :) I will try and take a look at this tomorrow or Monday

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,131 @@
+# Just three pages - license agreement, install location, and installation
+page license
+page directory
+Page instfiles
```
nit: Page vs page.

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,131 @@
+!include EnvVarUpdate.nsh
+ 
+# Just three pages - license agreement, install location, and installation
+page license
```
I would consider using MUI.  It will give you a nice UI and allow you to add easy branding and images.  It also give you internationalization for free for a built in messages.

http://nsis.sourceforge.net/Docs/Modern%20UI/Readme.html

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,131 @@
+# This installs two files, minikube.exe and logo.ico, creates a start menu shortcut, builds an uninstaller, and
```
More of a nit than anything.  NSIS functions are capitalized, 'Function', 'Section', etc.  They don't have an official style guide but this seems to be the standard.

iantalarico replied:
A good example of a nice MUI installer is notepad++.  It isn't to much to add you just need a few images and swap out your pages with the MUI ones.

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,131 @@
+	createShortCut "$SMPROGRAMS\${COMPANYNAME}\${APPNAME}.lnk" "$INSTDIR\minikube.exe" "" "$INSTDIR\logo.ico"
+ 
+	# Registry information for add/remove programs
+	WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${COMPANYNAME} ${APPNAME}" "DisplayName" "${COMPANYNAME} - ${APPNAME} - ${DESCRIPTION}"
```
Consider making "Software\Microsoft\Windows\CurrentVersion\Uninstall\${COMPANYNAME} ${APPNAME}" a variable.  I think it will make the below easier to read.

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,131 @@
+	SetShellVarContext all
+ 
+	#Verify the uninstaller - last chance to back out
+	MessageBox MB_OKCANCEL "Permanantly remove ${APPNAME}?" IDOK next
```
IIRC MUI has something built in for this if you decide to use it.

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,131 @@
+	rmDir "$SMPROGRAMS\${COMPANYNAME}"
+ 
+	# Remove files
+	delete $INSTDIR\minikube.exe
```
You can add the flag '/REBOOTOK' to RMDir and Delete.  They will be cleaned up after reboot if they couldn't be for some reason at the time of uninstall.

iantalarico commented:
Looks really good! I just added a few style comments and a suggestion for the UI.

aaron-prindle commented:
Thanks for your feedback @iantalarico!  I believe I have addressed all of the points you mentioned, please take a look whenever you get a chance.  Thanks!

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,176 @@
+# This installs two files, minikube.exe and logo.ico, creates a start menu shortcut, builds an uninstaller, and
+# adds uninstall information to the registry for Add/Remove Programs
+ 
+# To get started, put this script into a folder with the two files (minikube.exe, logo.ico, and license.rtf -
```
license.txt?

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,176 @@
+# To get started, put this script into a folder with the two files (minikube.exe, logo.ico, and license.rtf -
+# You'll have to create these yourself) and run makensis on it
+ 
+# If you change the names "minikube.exe", "logo.ico", or "license.rtf" you should do a search and replace - they
```
license.txt?

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,176 @@
+!define UNINSTALLDIR "Software\Microsoft\Windows\CurrentVersion\Uninstall\${COMPANYNAME} ${APPNAME}"
+ 
+# rtf or txt file - remember if it is txt, it must be in the DOS text format (\r\n)
+LicenseData "LICENSE.txt"
```
I don't think you need this with MUI_PAGE_LICENSE

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,176 @@
+# This will be in the installer/uninstaller's title bar
+Name "${COMPANYNAME} - ${APPNAME}"
+Icon "logo.ico"
+outFile "minikube-installer.exe"
```
nit: s/outFile/OutFile

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,176 @@
+!define MUI_LANGDLL_ALLLANGUAGES
+;Languages
+
+  !insertmacro MUI_LANGUAGE "English"
```
If you do all these languages I think you will need:

Unicode true

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,176 @@
+!include "MUI.nsh"       ; Modern UI
+ 
+ 
+# !insertmacro MUI_PAGE_WELCOME
```
I would suggest adding:
MUI_ICON and MUI_UNICON: Icon for the installer and uninstaller
MUI_WELCOMEFINISHPAGE_BITMAP and MUI_UNWELCOMEFINISHPAGE_BITMAP: Gives a nice branded feel to the installer first page
MUI_HEADERIMAGE_BITMAP:  Gives a nice branded feel to the installers other pages (hanldes uninstaller too)

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,176 @@
+ 
+InstallDir "$PROGRAMFILES\${COMPANYNAME}\${APPNAME}"
+!define UNINSTALLDIR "Software\Microsoft\Windows\CurrentVersion\Uninstall\${COMPANYNAME} ${APPNAME}"
+ 
```
Sorry I missed this before but I would also add:

BrandingText " "

This will remove the random branding text about null soft of the bottom of the installer.

iantalarico commented:
Just a few more small comments.  Nice work!

aaron-prindle commented:
I have made all suggested changes. PTAL @iantalarico, @dlorenc.  Thanks!

iantalarico commented on installers/windows/minikube.nsi:
```diff
@@ -0,0 +1,182 @@
+# If you change the names "minikube.exe", "logo.ico", or "LICENSE.txt" you should do a search and replace - they
+# show up in a few places.
+# All the other settings can be tweaked by editing the !defines at the top of this script
+# Unicode true # This command is not available in 2.46 which is on apt-get. Debating how to proceed
```
This will just cause some languages to not appear correctly.  This is probably okay to start with but I would be sure not to lose track of this (maybe file a bug) as it will be a bad UX for some users.

aaron-prindle commented:
@minikube-bot retest this please

dlorenc commented:
LGTM. Feel free to merge whenever, I think you're using this to test some Jenkins stuff now though so I'll leave it for you.

aaron-prindle commented:
Please re-LGTM @dlorenc

dlorenc commented on Makefile:
```diff
@@ -13,8 +13,12 @@
-VERSION ?= v0.10.0
-DEB_VERSION ?= 0.10-0
+VERSION_MAJOR ?= 0
+VERSION_MINOR ?= 9
```
Could you re-bump these?

aaron-prindle replied:
Done
